### PR TITLE
KH2: Hotfix update for game verison 1.0.0.9

### PR DESCRIPTION
**Pull request by JaredWeakStrike**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
fixing the address anchors to use the updated versions for Epic.
added steam support

## How was this tested?
Checked if the memory address regions match between versions.

### CODE REVIEW

**Comment by JaredWeakStrike** [comment]
> Have you ever thought about using Signature Scans for this? That might give you some more stability in the future. At least if the byte patterns are relatively stable and searchable in this game in general, which depends on language / compiler

I have thought of it but have no idea how I would do that with pymem. Most of the game's modding just uses hard addresses and thats what the client uses. I don't see them updating the game again and if they do it'll be what's happening now where I'm waiting on other things to update their hard addresses so I dont really see a reason to put in more effort

File: worlds/kh2/Client.py [hunk 1]
```diff
@@ -589,8 +615,9 @@ async def IsInShop(self, sellable):
         # if journal=-1 and shop = 5 then in shop
         # if journal !=-1 and shop = 10 then journal
 
-        journal = self.kh2_read_short(0x741230)
-        shop = self.kh2_read_short(0x741320)
+        #todo:get these addresses updated
```
Comment (NewSoupVi) [inline worlds/kh2/Client.py:618]: Is this done now?

**Review by NewSoupVi — CHANGES_REQUESTED** [review]
Changes seem sensible, this is mostly about testing. If ppl can confirm that they've played full seeds with both supported versions and they acted correctly, I'll merge

File: worlds/kh2/Client.py (hunk 1 above)
Comment (JaredWeakStrike) [reply worlds/kh2/Client.py:618]: yeah its done and people have been running on this version of the client for 4 days any not a single person has had issues with the updated memory addresses or steam support

**Review by Exempt-Medic — APPROVED** [review]
This works for me for both EG and Steam

**Review by black-sliver — APPROVED** [review]
I have no way of testing, but it basically only changes some numbers, and was tested.

I'd recommend reworking a couple of things at some point though:
* use proper python coding style - that is snake_case, not Pascal case.
* define the variables in the class body - that is `now: int`, `save: int`, etc. before `__init__`, but read below.
* maybe make the addresses / address switching more abstract, since currently you assign some numbers somewhere with comments of outdated addresses, that have no meaning, and then change to different numbers 200 lines later.
  * so maybe `self.addresses = stream_addresses`, and use `self.addresses.slot1` for the actual int, where steam_addresses and epic_addresses would be of type GameAddresses, which is NamedTuple.

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**